### PR TITLE
allow user to define the template used. …

### PR DESCRIPTION
**Pull request by geosword**
Make it more flexible, and the user can use yaml for the configuration if thats their preference

### CODE REVIEW

**Comment by geosword** [comment]
BTW, This is "needed" because the current process of converting to toml breaks when presented with the multiline configuration detailed here: https://vector.dev/highlights/2020-02-21-file-source-multiline-support/
There may be other test cases that fail, and I accept that fixing conversion to toml might a more logical action. However since vector supports yaml, and IMHO being able to specify which template to use is a best practice, it seems this would not allow a workaround the multiline issue mentioned above, but make the role more flexible too. Thanks in advance for your consideration.

**Comment by geosword** [comment]
I have just realised there is another way around the problem I was trying to address without needing to override the template / "fix" the toml rendering. Defining multiline this way works as expected with the toml layout:

```
  source_craft_weblog:
    type: file
    include:
      - "some_place/logs.log"
    multiline.start_pattern: "^[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}"
    multiline.mode: halt_before
    multiline.condition_pattern: "^[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}"
    multiline.timeout_ms: 1000
```

**Comment by geosword** [comment]
Ill leave it up to you if you want to merge the PR to allow the template to be defined. Thanks for a great role btw.

File: galaxy.yml [hunk 1]
```diff
@@ -3,9 +3,11 @@
 namespace: dzervas
 name: vector
 readme: README.md
+version: 0.1.4a
```
Comment (gaetanfl) [inline galaxy.yml]: I guess you wanted to put this in defaults/main.yml

File: roles/vector/defaults/main.yml [hunk 2]
```diff
@@ -1,3 +1,4 @@
+vector_template: vector.toml.j2
```
Comment (gaetanfl) [inline roles/vector/defaults/main.yml:1]: Setting this variable won't do anything unless you make some changes here https://github.com/dzervas/ansible-vector/blob/master/roles/vector/tasks/main.yml#L38

**Review by gaetanfl — CHANGES_REQUESTED** [review]
Hi,
nice idea to allow to choose which configuration format to use and thanks for your contribution.
I guess you forgot to commit some files. When files are added I'd review it again.

File: roles/vector/defaults/main.yml (hunk 2 above)
Comment (geosword) [reply roles/vector/defaults/main.yml:1]: Very true.. Not sure how I missed that.

File: galaxy.yml (hunk 1 above)
Comment (geosword) [reply galaxy.yml]: Not really, was just bumping the version. I've reverted it to the original

File: roles/vector/tasks/main.yml [hunk 3]
```diff
@@ -37,7 +37,7 @@
 
 - name: Copy config
   template:
-    src: templates/vector.toml.j2
+    src: "{{ vector_template }}"
     dest: /etc/vector/vector.toml
```
Comment (gaetanfl) [inline roles/vector/tasks/main.yml]: Maybe you want to change the destination too if you want to use yaml format

Comment (geosword) [reply roles/vector/tasks/main.yml]: "Tempus fugit" as they probably said in Rome... at some point...
Yes fair point, and one that's possibly a bit more complicated than first glance. I see two options...
1) Have the role user define the destination filename (or format) in an additional variable
2) Get ansible to strip the .j2 from the template filename and use whatever's left.
I've opted for 1. An extra variable, but its simpler and doesnt add any complexity to the unit tests.
see what you think. (commit is on its way)

**Review by gaetanfl — APPROVED** [review]
Thanks for contributing. I would have made the same choice regarding file name.

**Pull request merged**